run_experiment.py

```diff
@@ -1,24 +1,42 @@
 import argparse
 import json
 import torch
+import matplotlib.pyplot as plt
+import numpy as np
 
 from disvae.vae import VAE
-from utils.dataloaders import get_dataloaders
+from utils.datasets import get_dataloaders
 from viz.visualize import Visualizer as Viz
+from utils.load_model import load_model
 
 
-def random_samples(experiment_name, num_samples=1):
-    """ generate a number of random samples from the dataset
+def samples(experiment_name, num_samples=1, batch_size=1, shuffle=True):
+    """ generate a number of samples from the dataset
     """
     with open('experiments/{}/specs.json'.format(experiment_name)) as spec_file:
         spec_data = json.load(spec_file)
         dataset_name = spec_data['dataset']
 
-        _, test_loader = get_dataloaders(batch_size=num_samples, dataset=dataset_name, shuffle=True)
+        data_loader = get_dataloaders(batch_size=batch_size, dataset=dataset_name, shuffle=shuffle)
 
-        for (test_data, _) in test_loader:
-            return test_data
+        for batch_idx, (test_data, label) in enumerate(data_loader):
+            if num_samples == batch_idx:
+                break
+            if batch_idx == 0:
+                data = test_data
+            else:
+                data = torch.cat(tensors=(data, test_data))
```

Comment:
don't concatenate in loops : that's heavy.
Append to a list at every iteration (list append is O(1) in python)
then `torch.cat(list)`at the end of the loop.

If you want to be pythonic use `itertools.takewhile` and list comprehensions. But i's not curcuial at all

Reply:
I've just looked at the ```torch.cat()``` documentation in detail. I see the functionality is set up for multiple concatenations . Thanks for pointing this out!